### PR TITLE
[CBRD-23842] Classify DML statement performed by trigger action

### PR DESCRIPTION
**Pull request by joohok**
http://jira.cubrid.org/browse/CBRD-23842

This PR is to classify the DML logging that is appended by trigger action or not. 

Describe here

### CODE REVIEW

File: src/query/execute_statement.c [hunk 1]
```diff
@@ -16141,6 +16148,8 @@ check_merge_trigger (PT_DO_FUNC * do_func, PARSER_CONTEXT * parser, PT_NODE * st
       result = tr_before (state);
       if (result == NO_ERROR)
 	{
+	  cdc_Trigger_involved = is_trigger_involved;
```
Comment (joohok) [inline src/query/execute_statement.c]: Make macro for backup and restore the cdc_Trigger_involved

File: src/transaction/transaction_cl.c [hunk 2]
```diff
@@ -56,6 +56,7 @@
 #include "db.h"			/* for db_Connect_status */
 #include "porting.h"
 #include "network_interface_cl.h"
+#include "execute_statement.h"
```
Comment (joohok) [inline src/transaction/transaction_cl.c]: remove this part

File: src/object/work_space.h [hunk 3]
```diff
@@ -397,6 +398,22 @@ struct ws_statistics
     } \
   while (0)
 
+#define WS_ISTRIGGER(mop) ((mop)->trigger_involved)
```
Comment (hornetmj) [inline src/object/work_space.h]: How about WS_IS_TRIGGER_INVOLVED?

File: src/object/work_space.h [hunk 4]
```diff
@@ -397,6 +398,22 @@ struct ws_statistics
...
+
+#define WS_SET_TRIGGER(mop) \
```
Comment (hornetmj) [inline src/object/work_space.h]: How about WS_SET_TRIGGER_INVOLVED?

File: src/query/execute_statement.h [hunk 5]
```diff
@@ -58,6 +78,8 @@ typedef int (PT_DO_FUNC) (PARSER_CONTEXT *, PT_NODE *);
 
 extern bool do_Trigger_involved;
 
+extern bool cdc_Trigger_involved;
```
Comment (hornetmj) [inline src/query/execute_statement.h:81]: Please add comments why adding this global variable even though existing the do_Trigger_involved global variable.

File: src/query/query_cl.c [hunk 6]
```diff
@@ -116,6 +117,8 @@ execute_query (const XASL_ID * xasl_id, QUERY_ID * query_idp, int var_cnt, const
       return NO_ERROR;
     }
 
+  cdc_Trigger_involved = false;
```
Comment (hornetmj) [inline src/query/query_cl.c]: Please check prm_get_integer_value (PRM_ID_SUPPLEMENTAL_LOG).

File: src/query/execute_statement.h (hunk 5 above)
Comment (joohok) [reply src/query/execute_statement.h:81]: I added the comments above do_statement() in execute_statement.c where it is defined and initialized right below the do_Trigger_involved.

File: src/query/execute_statement.h [hunk 7]
```diff
@@ -33,6 +33,26 @@
 #include "dbi.h"
 #include "parser.h"
 
+#define CDC_TRIGGER_INVOLVED_BACKUP(is_trigger_involved) \
```
Comment (hornetmj) [inline src/query/execute_statement.h:36]: How about using a non-pointer type argument like CDC_TRIGGER_INVOLVED_RESTORE?

File: src/transaction/log_manager.c [hunk 8]
```diff
@@ -10762,15 +10768,43 @@ cdc_log_extract (THREAD_ENTRY * thread_p, LOG_LSA * process_lsa, CDC_LOGINFO_ENT
...
+		 * And this is a sequence for INSERT a record with OID reservation.
+		 * undo record with REC_ASSIGN_ADDRESS type has no undo image to extract, so this will be treated as INSERT
+		 * CUBRID engine used to do INSERT a record like this way,
+		 * for instance CREATE a class or INSERT a record by trigger execution */
+
+		assert (rec_type == LOG_SUPPLEMENT_TRIGGER_UPDATE);
+
+		error =
+		  cdc_make_dml_loginfo (thread_p, trid, tran_user,
+					rec_type == LOG_SUPPLEMENT_UPDATE ? CDC_INSERT : CDC_TRIGGER_INSERT, classoid,
```
Comment (hornetmj) [inline src/transaction/log_manager.c]:
```suggestion
					CDC_TRIGGER_INSERT, classoid,
```

**Comment by hornetmj** [comment]
@joohok I leave 2 additional comments. And I currently approve this PR. Please check them and merge this PR.

**Review by hornetmj — APPROVED** [review]

**Pull request merged**